### ISSUE TITLE
Display for relations when a relation is lost.

### ISSUE THREAD

**Issue by martinheidegger** [consento-org/mobile#9]
Both in "Consentos" and in "Locks" it is possible that each points to a relation that was deleted. We should show somehow the way that the user is displayed once the name is lost (it is still possible to unlock from a lost relation).

**Comment by marcchataigner-zz** [comment]
@martinheidegger good point. 

You mean, we are Consento relations, and then I lose/renew/reset my device. How does my contact appears on your device. Correct?

**Comment by martinheidegger** [comment]
> You mean, we are Consento relations, and then I lose/renew/reset my device. How does my contact appears on your device. Correct?

Actually no. when you lose/renew/reset the device there is currently no means to see former content. That is a different point.

When two people (alice & bob) create a relationsships, each device stores the communication settings to the other device: bob@alice and alice@bob

When alice adds bob@alice to her vault a lockee is created that that links to bob@alice. Also, bob receives a _consento_ that linked to the alice@bob relation.

Now, when alice deletes bob@alice it will not delete the lockee, as the vault may be locked at the time and we don't know what lockee's alice's vault has.

When alice now looks at the lockee, she will see a "unknown lockee" as the relationship is gone.

Bob has the consento and he can do a similar thing by deleting the alice@bob relationship. Once deleted he will still see the Consento pointing to a "unknown lockee" reference.

This "unknown lockee" display is what this issue is about.

I can think of two things that might be good to add here:

1. Always show the relationships id with the relationship (as the id will be the main link)
2. Create a default display to show what the consento/lockee looks like when the linked relation is gone.

**Comment by marcchataigner-zz** [comment]
I see. good point. 

What about

3. Alice cannot delete the relation without the consento of Bob. This would create a situation where Bob is notified and can find another lockee to replace Alice.

**Comment by martinheidegger** [comment]
> Alice cannot delete the relation without the consento of Bob. This would create a situation where Bob is notified and can find another lockee to replace Alice.

Nice, That is a good solution! It requires following things:

_(In my example alice is the one with the vault, so I continue to use that)_

1. A new consento that is shown at alice's device that bob is asking for the deletion of a relationship.
    - The consento has a display that shows how many vaults (locked with name  and unlocked without name) bob is currently a lockee to.
    - Only once bob is no lockee to any vault anymore it is possible for Alice to confirm the deletion.
2. A display at the bob's device that shows that alice received the deletion request and it is pending.
3. A means for bob to cancel the deletion request.

While this is a nice solution (and we should definitely implement that), it is a bit much work right now and I think a lost relationship display would be a cheap fix for now.

**Comment by marcchataigner-zz** [comment]
I agree. MVP as a goal first. 

As default, could it be that if Bob resigns from being a lockee, and is the only lockee of that vault, then the vault is set open by default? 

Up to Alice to find an other lockee? 
Or if the vault is sealed, then it cannot be set open without consent? 
(maybe on Bob's side resigning from being a lockee would include the message: "you'll make that vault open")?

**Comment by martinheidegger** [comment]
> As default, could it be that if Bob resigns from being a lockee, and is the only lockee of that vault, then the vault is set open by default?

We should not automatically unlock vaults. It is really easiest if we just have a "unknown relationship" display right now.

**Comment by marcchataigner-zz** [comment]
OK. And the relationship ID always looks like a long string of digits and letters, right?

**Comment by martinheidegger** [comment]
It is a bunch of random bytes than can be displayed in a variety of ways. Currently for the vault I use the first 8 bytes in a hex (0-9A-F) representation bundled with a `-` between two bytes:

`A67F-BB3C-1A89-23C4`

**Comment by martinheidegger** [comment]
After merging d720b59 I added the display in ac3492e. Closing this in favor of #15
